src/app/feeds.py

```diff
@@ -0,0 +1,166 @@
+import datetime
+import uuid
+from typing import Any, Optional
+
+import feedparser  # type: ignore[import-untyped]
+import PyRSS2Gen  # type: ignore[import-untyped]
+from flask import url_for
+
+from app import config, db, logger
+from app.models import Feed, Post
+from shared.podcast_downloader import find_audio_link
+
+
+def fetch_feed(url: str) -> feedparser.FeedParserDict:
+    logger.info(f"Fetching feed from URL: {url}")
+    feed_data = feedparser.parse(url)
+    for entry in feed_data.entries:
+        entry.id = get_guid(entry)
+    return feed_data
+
+
+def refresh_feed(feed: Feed) -> None:
+    logger.info(f"Refreshing feed with ID: {feed.id}")
+    feed_data = fetch_feed(feed.rss_url)
+    existing_posts = {post.guid for post in feed.posts}  # type: ignore[attr-defined]
+    oldest_post = min(
+        (post for post in feed.posts if post.release_date),  # type: ignore[attr-defined]
+        key=lambda p: p.release_date,
+        default=None,
+    )
+    for entry in feed_data.entries:
+        if entry.id not in existing_posts:
+            logger.debug(f"found new podcast: {entry.title}")
+            p = make_post(feed, entry)
+            if (
+                config.number_of_episodes_to_whitelist_from_archive_of_new_feed
+                is not None
+                and oldest_post
+                and p.release_date.date() < oldest_post.release_date
+            ):
```

Comment:
you're right... this was a mistake haha. Thinking a bit more about it though, this would act as a protection if the feed owner loads in a huge back catalog or something. 

what do you think about this modification?

    oldest_post = min(
        (post for post in feed.posts if post.release_date),  # type: ignore[attr-defined]
        key=lambda p: p.release_date,
        default=None,
    )
    for entry in feed_data.entries:
        if entry.id not in existing_posts:
            logger.debug(f"found new podcast: {entry.title}")
            p = make_post(feed, entry)
            # do not allow automatic download of any backcatalog added to the feed
            if (
                oldest_post is not None
                and p.release_date.date() < oldest_post.release_date
            ):

Reply:
Agh, it's all so messy :P 

I think if you want to go this road I might do `p.release_date.date() < newest_post.release_date` instead of `oldest_post`. 

You could also just retain the current behavior and call it something like `max_episodes_to_whitelist_one_pull` instead of `number_of_episodes_to_whitelist_from_archive_of_new_feed`. That behavior feels somehow a little easier to reason about at least for me. And that way you're also not relying on these timestamps which feels a bit dicey potentially. 🤷